### PR TITLE
chore(typings): add typings for withLatestFrom

### PR DESCRIPTION
**Pull request by david-driscoll**

### CODE REVIEW

File: src/operator/withLatestFrom.ts [hunk 1]
```diff
@@ -32,6 +32,27 @@ export function withLatestFrom<T, R>(...args: Array<Observable<any> | ((...value
...
+  <T2, T3, T4, R>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, project: (v1: T, v2: T2, v3: T3, v4: T4) => R): Observable<R>;
+  <T2, T3, T4, T5, R>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, project: (v1: T, v2: T2, v3: T3, v4: T4, v5: T5) => R): Observable<R>;
+  <T2, T3, T4, T5, T6, R>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>, project: (v1: T, v2: T2, v3: T3, v4: T4, v5: T5, v6: T6) => R): Observable<R>;
+
+  <T2>(v2: ObservableInput<T2>): Observable<[T, T2]>;
+  <T2, T3>(v2: ObservableInput<T2>, v3: ObservableInput<T3>): Observable<[T, T2, T3]>;
+  <T2, T3, T4>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>): Observable<[T, T2, T3, T4]>;
+  <T2, T3, T4, T5>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>): Observable<[T, T2, T3, T4, T5]>;
+  <T2, T3, T4, T5, T6>(v2: ObservableInput<T2>, v3: ObservableInput<T3>, v4: ObservableInput<T4>, v5: ObservableInput<T5>, v6: ObservableInput<T6>): Observable<[T, T2, T3, T4, T5, T6]>;
```
Comment (saneyuki) [inline src/operator/withLatestFrom.ts]: RxJS v4 defines ~`T9`: https://github.com/Reactive-Extensions/RxJS/blob/master/ts/core/linq/observable/withlatestfrom.ts. I feel we should define to it for the type comparibility.

**Comment by blesh** [comment]
@david-driscoll ... there are apparently build errors with this PR. Just a heads up in case you hadn't noticed.

**Comment by david-driscoll** [comment]
@blesh :dizzy_face:  Doh!

Fixed now.

**Comment by kwonoj** [comment]
Change looks ok to me. I'll leave this PR bit more to see if there's any other suggestion. For parity to RxJS4, as same as other definitions currently checked in I think it's ok for now instead of expand to ~T9. Those can be easily expanded as needed in further.

**Comment by kwonoj** [comment]
Merged with 7bbe33d, thanks @david-driscoll !

**Pull request closed without merging**